.github/workflows/autobuild.yaml

```diff
@@ -1,41 +1,124 @@
-# Workflow to build binaries for a release.
-# Triggered by release creation, which should include `llvm-project@<full-sha>`.
+# Workflow to build binaries and release them.
+# Triggered by the schedule or manual dispatch, which might include
+# `<owner/llvm-project>@<full-sha>`.
 #
 # Because the build takes more than an hour, our GITHUB_TOKEN credentials may
 # expire. A token `secrets.RELEASE_TOKEN` must exist with public_repo scope.
 name: Build release binaries
 on:
-  release:
-    types: created
+  # Run weekly on sunday at 21:37 UTC (arbitrary)
+  schedule:
+  - cron: '37 21 * * SUN'
+  workflow_dispatch:
+    inputs:
+      commit:
+        description: 'Commit to build from'
+        required: false
+      repo:
+        description: 'Repository to build from'
+        required: false
+        default: 'llvm/llvm-project'
+      tag:
+        description: 'Tag name'
+        required: false
 jobs:
-  prepare:
+  create_environment:
+    name: Create build environment
     runs-on: ubuntu-latest
     steps:
-    - name: Find LLVM commit
+    - name: Install deps
       run: |
-        perl -ne 'print "LLVM_REPO=$1\nLLVM_COMMIT=$2\n" if /(\w+\/llvm-project)@([[:xdigit:]]{40,})/' > commit.env <<EOF
-        ${{ github.event.release.body }}
-        EOF
-    - name: Check LLVM commit was founud
-      run: grep LLVM_REPO commit.env
-    - name: Mark as draft
+        sudo apt-get install jq
+    - name: Clone scripts
+      uses: actions/checkout@v2
+    - name: Pull repo from the inputs
```

Comment:
At the moment the idea of "config used for the periodic snapshot builds" seems conflated with "defaults when invoked explicitly". And these values are initialized throughout the workflow as they're needed rather than in one place.

Seems likely to be clearer to have a section at the front (as close so the trigger defs as possible) that initialized the environment based either on the passed inputs or with a fixed config for the scheduled builds. And this would be conditional on whether the trigger was schedule or workflow_dispatch, not on whether any particular field was empty.

Reply:
I've had this at the start but decided to merge this. Though unfortunate and seemingly not logical, the problem is that the logic is very similar in case some of the values are not filled in forth `workflow_dispatch` variables. I.e. when not filled in, the CI should still pull the defaults and all the options I've tried or thought about trying ended up being even more confusing and/or long than the one we have right now.

Reply:
names of these actions seems to be in the wrong order, you are using the default repo here, but have the `use the default repo` below.

also i'd rename to `Use repo from the inputs`

Reply:
> I've had this at the start but decided to merge this. Though unfortunate and seemingly not logical, the problem is that the logic is very similar in case some of the values are not filled in forth `workflow_dispatch` variables. I.e. when not filled in, the CI should still pull the defaults and all the options I've tried or thought about trying ended up being even more confusing and/or long than the one we have right now.

hmm, I think we can make it shorter/less complicated by assuming user will not provide any empty inputs on manual dispatches (and error out while creating the release when some env variable we need isn't set).